### DISCUSSION TITLE
Smooth shading does not smooth custom PolyData

### DISCUSSION THREAD

**Discussion by amilasilva92** [pyvista/pyvista#3030] [Q&A] [answered]
Hi, 

I want to smooth the surface of a custom obj file that has less number of vertices. I am using the following code snippet. However, it returns a non-smoothed 3d model as shown below. Could someone help me to solve this matter?
Thank you.

```
import pyvista as pv   
reader = pv.get_reader(obj_filename)
mesh = reader.read()
mesh = mesh.smooth(n_iter=50)
mesh.plot(show_edges=False, smooth_shading=True)
```

![Screen Shot 2022-07-19 at 12 50 07 pm](https://user-images.githubusercontent.com/25021490/179653799-849d7969-60f9-4ac5-9501-26336e96eaa3.png)

**Comment by akaszynski** [comment, marked as answer]
Could you please upload the mesh you used to generate this plot? I'm guessing you need to run [clean](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyData.clean.html) as the triangles are probably disconnected.

**Comment by amilasilva92** [comment]
Yes. It worked with clean function. Thank you so much for the help.
<img width="504" alt="Screen Shot 2022-07-20 at 7 50 36 pm" src="https://user-images.githubusercontent.com/25021490/179953256-e2a3df15-c7b8-47e4-8d09-4850d2a4a3c1.png">
